### PR TITLE
feat: add profiles filtering

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
Based on the provided task requirements and code changes, here is the review:

1. **app/page.tsx**: The `FiltersProvider` has been added to the component tree, which is necessary for providing filter context to the child components. This change is in line with the task requirements.

2. **src/components/Filters/Buttons/DevTypeButton/DevTypeButton.module.scss**: The new CSS class `.pressed` has been added to style the button when it's pressed. This change is in line with the task requirements.

3. **src/components/Filters/Buttons/DevTypeButton/DevTypeButton.tsx**: The `isPressed` prop has been added to the `DevTypeButton` component to handle the button press state. This change is in line with the task requirements.

4. **src/components/Filters/Filters.tsx**: The filters have been updated to handle single selections instead of multiple selections, and filter state is now managed using the `useFilters` hook from the `FilterContext`. The `handleSelect` function has been updated to handle the selection of filters and update the corresponding filter state. The `handleButtonClick` function has been added to handle the selection of stack type (Frontend, Backend, Fullstack). This change is in line with the task requirements.

5. **src/components/ProfileList/ProfileData.ts**: The profile data structure has been updated to include more details and align with the filter options. This change is in line with the task requirements.

6. **src/components/ProfileList/ProfileList.module.scss**: Minor style changes have been made to better present the technology tags. This change is in line with the task requirements.

7. **src/components/ProfileList/ProfileList.tsx**: The `ProfileList` component has been updated to filter profiles based on selected filters using the `useFilters` hook from the `FilterContext`. This change is in line with the task requirements.

8. **src/contexts/FilterContext.tsx**: A new context `FilterContext` has been created to manage filter state across components. This change is in line with the task requirements.

9. **src/inputs/Dropdowns/DropdownFilter/DropdownFilter.tsx**: The `DropdownFilter` component has been updated to handle single selections instead of multiple selections. This change is in line with the task requirements.

10. **src/inputs/Dropdowns/DropdownFilterMulti/DropdownFilterMulti.tsx**: A new component `DropdownFilterMulti` has been created to handle multiple selections for the technology filter. This change is in line with the task requirements.

In conclusion, all code changes are in line with the task requirements. There are no apparent bugs or issues in the provided code changes.

File: app/page.tsx [hunk 1]
```diff
@@ -2,14 +2,15 @@ import React from 'react'
 import Filters from '@/components/Filters/Filters'
 import DefaultHeader from '@/components/Headers/DefaultHeader/DefaultHeader'
 import ProfileList from '@/components/ProfileList/ProfileList'
+import { FiltersProvider } from '@/contexts/FilterContext'
 
 const Home: React.FC = () => {
   return (
-    <>
+    <FiltersProvider>
```
Comment (Sumick) [inline app/page.tsx:9]: **praise:** Good choice for solving this problem 👌

File: src/components/ProfileList/ProfileData.ts [hunk 2]
```diff
@@ -2,72 +2,141 @@ export interface ProfileListItems {
...
-  technology3: string
+  stack: string
   others: string
+  country: string
+  city: string
+  remote: string
+  technology: string[]
+  employmentType: string
+}
+
```
Comment (Sumick) [inline src/components/ProfileList/ProfileData.ts:13]: **nitpick:** It's not a component so the file should be named with kebab case: `profile-data.ts`

File: src/components/ProfileList/ProfileData.ts [hunk 3]
```diff
@@ -2,72 +2,141 @@ export interface ProfileListItems {
...
+    others: '+5 more',
+    country: 'Spain',
+    city: 'Barcelona',
+    remote: 'Remote',
+    technology: getRandomTechnologies(),
+    employmentType: 'Part-time',
+  },
+  {
+    id: 9,
+    name: 'Krzysztof Misiorny',
```
Comment (Sumick) [inline src/components/ProfileList/ProfileData.ts:132]: **praise:** xD

File: src/components/ProfileList/ProfileList.tsx [hunk 4]
```diff
@@ -1,79 +1,52 @@
...
-  return <p className={`${styles.badge} ${themeClass}`}>{props.name}</p>
-}
+import { useFilters } from '@/contexts/FilterContext'
+import classNames from 'classnames/bind'
+const cx = classNames.bind(styles)
 
 const ProfileListItem: React.FC<{ data: ProfileListItems }> = ({ data }) => {
-  const [theme, setTheme] = useState<JobSpecialization>(JobSpecialization.None)
+  const getStackClasses = cx({
+    [styles.frontend]: data.stack === 'Frontend',
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:15]: **nitpick:** You already duplicated these values few times, and you have enum for that: `JobSpecialization.Frontend`, let's use it everywhere

File: src/components/ProfileList/ProfileList.tsx [hunk 5]
```diff
@@ -82,42 +55,42 @@ const ProfileListItem: React.FC<{ data: ProfileListItems }> = ({ data }) => {
...
 
 const ProfileList: React.FC = () => {
+  const {
+    stackFilter,
+    technologyFilter,
+    seniorityFilter,
+    availabilityFilter,
+    locationFilter,
+  } = useFilters()
+  const filteredProfileData = ProfileData.filter((profile) => {
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:79]: **nitpick:** `ProfileData` should be written with camelCase (`profileData`)

File: src/components/ProfileList/ProfileList.tsx [hunk 6]
```diff
@@ -82,42 +55,42 @@ const ProfileListItem: React.FC<{ data: ProfileListItems }> = ({ data }) => {
...
+    return (
+      (!stackFilter || profile.stack === stackFilter) &&
+      (!seniorityFilter || profile.seniority === seniorityFilter) &&
+      (!locationFilter || profile.country === locationFilter) &&
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:83]: **nitpick:** We should split this, as it's non maintainable, and quite messy. Let's chain these filters, so it's cleanly understandable which filter does what. Let's create filter chain as following:

```js
const filteredProfileData = profileData.filter(filterByStack(stackFilter)).filter(filterBySeniority(seniorityFilter))
...
```

**Review by Sumick — CHANGES_REQUESTED** [review]
Starting to look good - we need few improvements

**Review by hansac91 — APPROVED** [review]
